### ISSUE TITLE
wazuh-ruleset: Cisco rules and decoders improvements

### ISSUE THREAD

**Issue by juanrricci** [wazuh/wazuh#7278]
This issue is associated to the following PR that belong to wazuh-ruleset: https://github.com/wazuh/wazuh-ruleset/pull/402

The aim of this issue is to solve the plausible conflicts that could raise while merging with the current ruleset repository. The associated PR is older than a year and refers to several issues with the Cisco ruleset.

**Comment by juanrricci** [comment]
Replication of the issue https://github.com/wazuh/wazuh-ruleset/issues/269 (associated with the main PR) in Centos7 with Wazuh Manager v4.2.0.

Testing PIX logs with `wazuh-logtest`:
PIX decoder file: `0210-pix_decoders.xml`
Example logs (taken from the PIX decoder file): 
```
%PIX-6-106015: Deny TCP (no connection) from 161.58.238.151/110 to a.b.c.d/3782 flags RST ACK
%PIX-2-106001: Inbound TCP connection denied from 165.139.46.7/3854 to 165.189.27.70/139 flags
%PIX-3-106010: Deny inbound tcp src outside:213.98.79.233/2620 dst dmz:213.98.254.145/135
%PIX-3-106011: Deny inbound (No xlate) udp src outside:192.168.2.1/137
%PIX-3-106011: Deny inbound (No xlate) tcp src inside:10.100.7.43/80 dst
%PIX-3-710003: TCP access denied by ACL from 216.39.220.130/54065 to outside:62.192.113.98/ssh
%PIX-7-710001: TCP access requested from X.X.X.X/1292 to outside:Y.Y.Y.Y/ssh
%PIX-7-710002: UDP access permitted from 33.33.33.4/943 to inside:33.33.33.15/snmp
%PIX-7-710005: UDP request discarded from <public IP of 525>/4500 to outside:192.168.69.137/4500
%PIX-2-106002   protocol Connection denied by outbound list acl_ID src inside_address dest outside_address
%PIX-2-106002: udp connection denied by outbound list 30 src 216.53.120.62 138 dest 169.132.10.82 138
 %PIX-4-106023: Deny tcp src inside:111.11.11.1/2143 dst YYY:172.11.1.11/139 by access-group "inside_inbound"
%PIX-4-400013 IDS:2003 ICMP redirect from 10.4.1.2 to 10.2.1.1 on interface dmz
%PIX-2-106006: Deny inbound UDP from ***/20031 to ***/20031 on
%PIX-7-710002: TCP access permitted from 10.0.0.1/60749 to db:10.0.0.2/ssh
%PIX-6-305012: Teardown dynamic UDP translation from inside:1.1.1.1/12 to outside:1.2.1.2/11 duration 0:00:11.
%PIX-3-305005: No translation group found for icmp src outside:x.x.x.x dst inside:x.x.x.x (type 3, code 0)
%ASA-2-106001: Inbound TCP connection denied from 1.2.3.4/1234 to 213.207.99.248/445 flags SYN on interface outside (Message repeated 2 times)
%PIX-6-605005: Login permitted from 192.168.1.2/2953 to inside:192.168.1.1/telnet for user ""
%PIX-6-605004: Login denied from 192.168.2.10/32597 to outside:192.168.2.14/ssh for user "root"
%PIX-6-305011: Built dynamic UDP translation from inside:192.168.1.2/1026 to outside:192.168.2.14/1163
%PIX-6-305011: Built dynamic TCP translation from inside:192.168.1.3/54946 to outside:192.168.2.14/1033
%PIX-6-302015: Built outbound UDP connection 156 for outside:192.168.2.10/1514 (192.168.2.10/1514) to inside:192.168.1.2/1026 (192.168.2.14/1163)
```
Matched decoder: most of them match with `cisco-ios` (located in `0065-cisco-ios_decoders.xml`)
Proposed solution: change PIX decoder file name from `0210-pix_decoders.xml` to `0064-pix_decoders.xml` for it to take precedence over the `cisco-ios` decoders.
Results: after renaming the PIX decoder file and restarting the wazuh-manager service, most of the example logs matched properly the PIX decoder but some of them do not match with the rule `4300` (`Grouping of PIX rules` located in `0065-pix_rules.xml`) or any of its children rules but match with the rule `4100` (`Firewall rules grouped` located in `0060-firewall_rules.xml`).
New proposed solution:
- Change PIX decoder file name from `0210-pix_decoders.xml` to `0064-pix_decoders.xml` for it to take precedence over the `cisco-ios` decoders.
- Verify if the example logs are still valid.
- Constrain the firewall rules for avoiding matching with logs decoded by the PIX decoder.

**Comment by juanrricci** [comment]
New proposed solution - update:
- Change PIX decoder file name from 0210-pix_decoders.xml to 0063-pix_decoders.xml for it to take precedence over the cisco-ios decoders and avoid overriding `0064-cisco-asa_decoders.xml`.
- Add missing PIX child decoders for logs that only pre-match with the parent decoder.
- Add `<type>firewall</type>` in child PIX decoders and `<category>firewall</category>` in parent PIX rule `4300` for proper categorization.
- Move the `0060-firewall_rules.xml` to the last position in the ruleset (for example by renaming the rule file to `0900-firewall_rules.xml`) for avoid the generic firewall rules taking precedence over firewall rules for specific vendors.

**Comment by juanrricci** [comment]
Replication of issues associated to PR https://github.com/wazuh/wazuh-ruleset/pull/402 in Centos7 with Wazuh Manager v4.2.0.

Changes on rules and decoders:
- Removed lines   
  `<prematch>^%ASA-|^\w\w\w \d\d \d\d\d\d \d\d:\d\d:\d\d: %ASA-|</prematch>`
  `<prematch>^%FWSM-|^\w\w\w \d\d \d\d\d\d \d\d:\d\d:\d\d: %FWSM-</prematch>` 
  from `0210-pix_decoders.xml` prematch.
- Substituted `<prematch>^%ASA-</prematch>`
  for `<prematch>%ASA-|\w\w\w \d\d \d\d\d\d \d\d:\d\d:\d\d: %ASA-</prematch>`
  in `0064-cisco-asa_decoders.xml`.
- Substituted current decoder file `0065-cisco-ios_decoders.xml` for the one provided by the owner of the related PR with some minimal changes for solving the related issues properly. The new decoder file includes many examples and explanations about Cisco log formats.
- Substituted current rule file `0075-cisco-ios_rules.xml` for the one provided by the owner of the related PR for proper log severity / rule level matching.
- Added example logs provided in related issues to `cisco_ios.ini` test file. All tests pass OK.
- Added example log provided in related issues to `cisco_asa.ini` test file. All tests pass OK.

**Comment by juanrricci** [comment]
Changes related to PR https://github.com/wazuh/wazuh-ruleset/pull/271:
- Applied changes for:
  `0185-openldap_decoders.xml`
  `0455-pfsense_decoders.xml`
  `0020-syslog_rules.xml`
  `0305-dropbear_rules.xml`
- Discarded changes for `0015-ossec_rules.xml`
- Added new test files:
  `dropbear.ini`
  `openldap.ini`
  `pfsense.ini`

**Comment by juanrricci** [comment]
Changes related to PR https://github.com/wazuh/wazuh-ruleset/pull/443:
- Added the following files with minimal changes for compatibility:
  `0099-f5_bigip_decoders.xml`
  `0695-f5_bigip_rules.xml`
  `f5_bigip.ini`

**Comment by juanrricci** [comment]
Changes related to PR https://github.com/wazuh/wazuh-ruleset/pull/444:
- Suggested changes in `0299-sophos_decoders.xml` added to `0300-sophos_decoders.xml`.
- Suggested changes in `0695-sophos_rules.xml` added to `0415-sophos_rules.xml`.
- Added `sophos.ini` test file with several example logs. All tests pass OK.

PR closed.

**Comment by juanrricci** [comment]
Changes related to PR https://github.com/wazuh/wazuh-ruleset/pull/510:
- Suggested changes added to `0040-auditd_decoders.xml`.
- Added auditd.ini test file. All tests pass OK.

PR closed.

**Comment by juanrricci** [comment]
Changes related to PR https://github.com/wazuh/wazuh-ruleset/pull/519:
- Renamed `0500-sophos_fw_decoders.xml` as `0510-sophos_fw_decoders.xml` and added to the decoder set.
- Renamed `0690-sophos_fw_rules.xml` as `0705-sophos_fw_rules.xml` and added to the rule set.
- Added `sophos_fw.ini` test file. All tests pass OK.

PR closed.

**Comment by juanrricci** [comment]
Changes related to PR https://github.com/wazuh/wazuh-ruleset/pull/505/files:
- Changed `win_authentication_failed` group and group references to `authentication_failed`.
- Removed rule `18152` from `0220-msauth_rules.xml`.

PR closed.

**Comment by juanrricci** [comment]
Changes related to PR https://github.com/wazuh/wazuh-ruleset/pull/334:
- Renamed `0500-wazuh-api-decoders.xml` to `0515-wazuh-api-decoders.xml` and added to the decoder set.
- Renamed `0590-wazuh-api-rules.xml`  to `0710-wazuh-api-rules.xml` and added to the rule set.
- Added `api.ini` test file. All tests pass OK.

PR closed.

**Comment by juanrricci** [comment]
Ran all test files. Found that `0099-f5_bigip_decoders.xml` have pretty general regular expressions for matching logs. We did not figure out how to properly set the prematch regex (it looks like the pre-decoding phase prevents from matching the timestamp and hostname portion of logs) so we decided on moving these decoders to the end of the decoder set by renaming it as `0520-f5_bigip_decoders.xml` as a workaround.
All tests passed properly after making this change.
